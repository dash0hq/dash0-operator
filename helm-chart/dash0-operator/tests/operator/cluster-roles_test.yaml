suite: test cluster roles
templates:
  - operator/cluster-roles.yaml
tests:
  - it: cluster roles should match snapshot
    asserts:
      - matchSnapshot: {}