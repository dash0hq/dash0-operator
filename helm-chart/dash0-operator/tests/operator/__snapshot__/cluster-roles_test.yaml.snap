cluster roles should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/component: kube-rbac-proxy
        app.kubernetes.io/instance: manager-role
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: clusterrole
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-manager-role
    rules:
      - apiGroups:
          - apps
        resources:
          - daemonsets
          - deployments
          - replicasets
          - statefulsets
        verbs:
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - batch
        resources:
          - cronjobs
          - jobs
        verbs:
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - list
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - delete
          - get
          - list
      - apiGroups:
          - operator.dash0.com
        resources:
          - dash0s
        verbs:
          - create
          - delete
          - deletecollection
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - operator.dash0.com
        resources:
          - dash0s/finalizers
        verbs:
          - update
      - apiGroups:
          - operator.dash0.com
        resources:
          - dash0s/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - configmaps
          - serviceaccounts
          - services
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - rbac.authorization.k8s.io
        resources:
          - clusterroles
          - clusterrolebindings
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - apps
        resources:
          - daemonsets
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - pods
          - namespaces
        verbs:
          - get
          - watch
          - list
      - apiGroups:
          - apps
        resources:
          - replicasets
        verbs:
          - get
          - watch
          - list
      - apiGroups:
          - extensions
        resources:
          - replicasets
        verbs:
          - get
          - watch
          - list
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/component: kube-rbac-proxy
        app.kubernetes.io/instance: metrics-reader
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: clusterrole
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-metrics-reader
    rules:
      - nonResourceURLs:
          - /metrics
        verbs:
          - get
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/component: kube-rbac-proxy
        app.kubernetes.io/instance: proxy-role
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: clusterrole
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-proxy-role
    rules:
      - apiGroups:
          - authentication.k8s.io
        resources:
          - tokenreviews
        verbs:
          - create
      - apiGroups:
          - authorization.k8s.io
        resources:
          - subjectaccessreviews
        verbs:
          - create
